usecases/objects/references_add.go

```diff
@@ -59,26 +58,32 @@ func (m *Manager) AddObjectReference(ctx context.Context, principal *models.Prin
 		return &Error{"cannot lock", StatusInternalServerError, err}
 	}
 	defer unlock()
-
 	validator := validation.New(m.vectorRepo.Exists, m.config, repl)
-	if err := input.validate(ctx, principal, validator, m.schemaManager, tenant); err != nil {
+	targetRef, err := input.validate(principal, validator, m.schemaManager)
+	if err != nil {
 		if errors.As(err, &ErrMultiTenancy{}) {
 			return &Error{"validate inputs", StatusUnprocessableEntity, err}
 		}
 		return &Error{"validate inputs", StatusBadRequest, err}
 	}
 
-	if !deprecatedEndpoint {
-		if input.Class != "" && strings.Count(string(input.Ref.Beacon), "/") == 3 {
-			toClass, toBeacon, replace, err := m.autodetectToClass(ctx, principal, input.Class, input.Property, input.Ref.Beacon)
-			if err != nil {
-				return err
-			}
-			if replace {
-				input.Ref.Class = toClass
-				input.Ref.Beacon = toBeacon
-			}
+	if input.Class != "" && targetRef.Class == "" {
```

Comment:
you have removed here `if !deprecatedEndpoint` check. I wonder if this change affects the logic?

Reply:
ohh it's moved a little bit to the bottom

Reply:
deprecatedEndpoint is `input.Class == "",` so it was redundant. I found using `input.Class != ""` clearer than using `deprecatedEndpoint` because this piece of code is not about the old endpoint, but about simplifying calls